index.bs

```diff
@@ -1993,6 +1995,116 @@ The [=remote end steps=] with |command parameters| are:
+
+      BrowsingContextReloadParameters = {
+        context: BrowsingContext,
+        delta: int,
+        ?wait: ReadinessState,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        BrowsingContextTraverseHistoryResult = {
+            navigation: Navigation / null,
+            ?persisted: bool
+            url: text,
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.traverseHistory">
+The [=remote end steps=] with |command parameters| are:
+
+  1. Let |context id| be the value of the <code>context</code> field of
+     |command parameters|.
+
+  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+     with |context id|.
+
+  1. Assert: |context| is not null.
+
+  1. Let |delta| be the value of the <code>delta</code> field of |command
+     parameters|.
+
+  1. Let |wait condition| be the value of the <code>wait</code> field of |command
+     parameters| if present, or "<code>none</code>" otherwise.
+
+  1. Let |navigate status| be the result of running the [=traverse the history
+     by a delta steps=] given |delta|, and |context|.
+
+  1. If |navigate status|'s status is "<code>canceled</code>" return [=error=]
+     with [=error code=] [=unknown error=].
+
+  1. Assert: |navigate status|'s status is "<code>pending</code>" and
+     |navigation id| is not null.
+
+  1. If |wait condition| is "<code>none</code>":
+
+   1. Let |body| be a [=map=] matching the
+     <code>BrowsingContextNavigateResult</code> production, with the
+     <code>navigation</code> field set to |navigation id|, and the
+     <code>url</code> field set to the result of the [=URL serializer=] given
+     |navigate status|'s url.
+
+ 1. If |wait condition| is "<code>interactive</code>", let |event name| be
+     "<code>domContentLoaded</code>", otherwise let |event name| be
+     "<code>load</code>".
+
+ 1. Let (|event received|, |status|) be [=await=] given «|event name|,
+    "<code>download started</code>", "<code>navigation failed</code>",
+    "<code>page show</code>"» and |navigation id|.
+
+ 1. If |event received| is "<code>navigation failed</code>"
+    return [=error=] with [=error code=] [=unknown error=].
+
+ 1. If |event received| is "<code>page show</code>", let |persisted| be true,
+    otherwise let |persisted| be false.
+
+  1. Let |body| be a [=map=] matching the
+     <code>BrowsingContextTraverseHistoryResult</code> production, with the
+     <code>navigation</code> field set to |status|'s id, the
+     <code>persisted</code> field set to |persisted|, and the <code>url</code>
+     field set to the result of the [=URL serializer=] given |status|'s url.
+
+  1. Return [=success=] with data |body|.
+
+</div>
+
+<div algorithm>
+
+The <dfn export>WebDriver-BiDi page show</dfn> steps given <var
+ignore>context</var> and |navigation status| are:
+
+Issue: Do we want to expose a `browsingContext.pageShow event? In that case we'd
```

Comment:
Isn't that what we will do for navigation anyway when no wait parameter has been given and the command returns immediately?

Reply:
The question is whether we want a specific event for it so that clients can wait for pageShow even if it's triggered by something other than a navigation command. I'm inclined to think it would probably be a good idea, given pageshow/pagehide are typically the most reliable things to use in the face of bfcache. I don't quite know if it's covered by `Page.lifecycleEvent` in CDP since the docs don't mention when the events are actually emitted.